pkg/storage/tsdb/caching_bucket.go

```diff
@@ -0,0 +1,89 @@
+package tsdb
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thanos-io/thanos/pkg/cache"
+	"github.com/thanos-io/thanos/pkg/cacheutil"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	storecache "github.com/thanos-io/thanos/pkg/store/cache"
+)
+
+type ChunksCacheConfig struct {
+	Backend   string                `yaml:"backend"`
+	Memcached MemcachedClientConfig `yaml:"memcached"`
+
+	// Basic unit used to cache chunks.
+	SubrangeSize        int64 `yaml:"subrange_size"`
+	MaxGetRangeRequests int   `yaml:"max_get_range_requests"`
+
+	// TTLs for various cache items.
+	ObjectSizeTTL time.Duration `yaml:"object_size_ttl"`
+	SubrangeTTL   time.Duration `yaml:"subrange_ttl"`
+}
+
+func (cfg *ChunksCacheConfig) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
+	f.StringVar(&cfg.Backend, prefix+"backend", "", fmt.Sprintf("Backend for chunks cache. Supported values: %s.", storecache.MemcachedBucketCacheProvider))
+
+	cfg.Memcached.RegisterFlagsWithPrefix(f, prefix+"memcached.")
+
+	defaultCfg := storecache.DefaultCachingBucketConfig()
```

Comment:
[not a blocker] Is it a good think dynamically picking defaults from Thanos instead of hard-coding them? I'm asking it because once we'll move blocks storage from experimental to stable, we'll have to avoid any breaking change both in config and defaults: being these defaults dynamic, a change in Thanos will mean a change to Cortex defaults too.

Reply:
I've hardcoded the values now (using defaults from Thanos). Following default values Thanos by reusing their "default" function could surprise us, so let's not do that.